modules/m_names.c

```diff
@@ -172,14 +172,33 @@ names_global(struct Client *source_p)
 		if(dont_show)
 			continue;
 
-		if((cur_len + NICKLEN + 2) > (BUFSIZE - 3))
+		if (IsCapable(source_p, CLICAP_USERHOST_IN_NAMES))
 		{
-			sendto_one(source_p, "%s", buf);
-			cur_len = mlen;
-			t = buf + mlen;
+			/* !@, space */
+			if (cur_len + strlen(target_p->name) + strlen(target_p->username) + strlen(target_p->host) + 3 >= BUFSIZE - 3)
```

Comment:
As mentioned on IRC, compilers will convert strlen of string literals into constants, so we can avoid using the magic constants ourselves:

```suggestion
			if (cur_len + strlen(target_p->name) + strlen(target_p->username) + strlen(target_p->host) + strlen(" !@") >= BUFSIZE - 3)
```